src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1771,13 +1602,8 @@ internal Document WithFrozenSourceGeneratedDocument(SourceGeneratedDocumentIdent
         /// </summary>
         internal Solution WithoutFrozenSourceGeneratedDocuments()
         {
-            var newState = _state.WithoutFrozenSourceGeneratedDocuments();
-            if (newState == _state)
-            {
-                return this;
-            }
-
-            return new Solution(newState);
+            var newCompilationState = _compilationState.WithoutFrozenSourceGeneratedDocuments(_state.GetProjectDependencyGraph());
```

Comment:
Why is this explicitly taking a dependency graph rather than doing that in the call off CompilationState itself?

Reply:
Good catch!  And it's even weirder.  We're passing in information from `_state`.  But `_compilationState` *already* points at `_state` (remember, the latter is justa  property that calls *through* `_compilationState` .  So we can remove the argument *entirely*!

Reply:
have done this.